mailbox/store/src/main/java/org/apache/james/mailbox/store/search/ListeningMessageSearchIndex.java

```diff
@@ -110,6 +110,18 @@ private void handleAdded(MailboxSession session, Mailbox mailbox, Added added) {
         }
     }
 
+    private void handleFlagsUpdated(MailboxSession session, Mailbox mailbox, List<UpdatedFlags> updatedFlagsList) {
```

Comment:
This, is what ElasticSearch should do.

There is no reasons to enforce this choice on other technologies. It will be suboptimal for Lucene.